src/Parallel/Algorithm.hpp

```diff
@@ -424,6 +464,49 @@ class AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>
     return number_of_actions;
   }
 
+  // Invoke the static `apply` method of `ThisAction`. The if constexprs are for
+  // handling the cases where the `apply` method returns a tuple of one, two,
+  // or three elements, in order:
+  // 1. A DataBox
+  // 2. A bool determining whether or not to terminate (and potentially move
+  //   to the next phase)
+  // 3. An unsigned integer corresponding to which action in the current phase's
+  //    algorithm to execute next.
+  template <typename ThisAction, typename ActionList, typename DbTags>
+  void invoke_iterable_action(db::DataBox<DbTags>& my_box) noexcept {
+    auto action_return = ThisAction::apply(
+        my_box, inboxes_, *global_cache_, std::as_const(array_index_),
+        ActionList{}, std::add_pointer_t<ParallelComponent>{});
+
+    static_assert(
+        detail::check_iterable_action_return_type<
+            ParallelComponent, ThisAction,
+            std::decay_t<decltype(action_return)>>::value,
+        "An iterable action returns an invalid return type."
+        "See the template parameters for details: the first is the parallel"
+        "component in question, the second is the iterable action, and the"
+        "third is the return type at fault."
+        "The return type must be a tuple of length one, two, or three "
+        "with:"
+        " first type is an updated DataBox;"
+        " second type is either a bool (indicating termination) or a "
+        "`AlgorithmControl` (directing algorithm control-flow); and"
```

Comment:
is AlgorithmControl in yet? might be better to wait to modify the text for that PR